### DISCUSSION TITLE
give me reason why do i need to use v3 instead of v2?

### DISCUSSION THREAD

**Discussion by dellwatson** [0xPolygon/matic.js#339] [General]
In my project, im currently using v2 because somehow it's working better with metamask and v3 implementation in browser client seems way off? I have no idea how to implement it correctly and by far it only able to use connectWallet instead of metamask too.

but what's the major difference? is there any kind of bug in v2 that v3 can truly solve? i have read the update but seems it's just minor update?

**Comment by ujjwalgupta94** [comment]
There were so many improvements done in v3

* you can use web3.js or ether.js based on your choice.
* The package is now light weight, the package does not have ABI hardcoded, it uses abi from server - load based on requirement. Particularly when you are developing web app, matic.js was contributing a big chunk.
* Better error handling - you will see lots of user friendly error message which will help you to know what has happened and how this can be fixed.
* Multiple helper api added, e.g - isCheckPointed, isDeposited etc.
* Direct access to build proof

V2 didn't had any major issue - it was working fine. But in order to achieve the above requirements , we had to create v3. I hope this helps you to understand v3 better.
